api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -694,6 +698,16 @@ public void updateWorkspaceBillingAccount(DbWorkspace workspace, String newBilli
       cloudbilling = serviceAccountCloudbillingProvider.get();
     } else {
       cloudbilling = endUserCloudbillingProvider.get();
+      try {
+        if (!Optional.ofNullable(
```

Comment:
That's actually a different statement since `isPresent()` would return true if `isOpen` is set to false.

But probably not a good sign if that's not clear so I refactored it a little bit by pulling out a variable.